File: scala/src/main/scala/com/spoonacular/client/model/InlineResponse20022.scala
package com.spoonacular.client.model

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import spoonacular._
import com.spoonacular.client.model.InlineResponse20022Nutrition
import com.spoonacular.client.model.RecipesParseIngredientsEstimatedCost
import java.math.BigDecimal
import scala.collection.immutable.Seq

/**
 * 
 * @param id 
 * @param original 
 * @param originalName 
 * @param name 
 * @param nameClean 
 * @param amount 
 * @param unit 
 * @param unitShort 
 * @param unitLong 
 * @param possibleUnits 
 * @param estimatedCost 
 * @param consistency 
 * @param shoppingListUnits 
 * @param aisle 
 * @param image 
 * @param meta 
 * @param nutrition 
 * @param categoryPath 
 */
case class InlineResponse20022(id: Int,
                original: String,
                originalName: String,
                name: String,
                nameClean: String,
                amount: BigDecimal,
                unit: String,
                unitShort: String,
                unitLong: String,
                possibleUnits: Seq[String],
                estimatedCost: RecipesParseIngredientsEstimatedCost,
                consistency: String,
                shoppingListUnits: Seq[String],
                aisle: String,
                image: String,
                meta: Seq[Object],
                nutrition: InlineResponse20022Nutrition,
                categoryPath: Seq[String]
                )

object InlineResponse20022 {
    /**
     * Creates the codec for converting InlineResponse20022 from and to JSON.
     */
    implicit val decoder: Decoder[InlineResponse20022] = deriveDecoder
    implicit val encoder: ObjectEncoder[InlineResponse20022] = deriveEncoder
}

File: clojure/src/spoonacular_api/specs/inline_response_200_22.clj
(ns spoonacular-api.specs.inline-response-200-22
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [spoonacular-api.specs.-recipes-parse-ingredients-estimated-cost :refer :all]
            [spoonacular-api.specs.inline-response-200-22-nutrition :refer :all]
            )
  (:import (java.io File)))


(def inline-response-200-22-data
  {
   (ds/req :id) int?
   (ds/req :original) string?
   (ds/req :originalName) string?
   (ds/req :name) string?
   (ds/req :nameClean) string?
   (ds/req :amount) float?
   (ds/req :unit) string?
   (ds/req :unitShort) string?
   (ds/req :unitLong) string?
   (ds/req :possibleUnits) (s/coll-of string?)
   (ds/req :estimatedCost) -recipes-parse-ingredients-estimated-cost-spec
   (ds/req :consistency) string?
   (ds/req :shoppingListUnits) (s/coll-of string?)
   (ds/req :aisle) string?
   (ds/req :image) string?
   (ds/req :meta) (s/coll-of any?)
   (ds/req :nutrition) inline-response-200-22-nutrition-spec
   (ds/req :categoryPath) (s/coll-of string?)
   })

(def inline-response-200-22-spec
  (ds/spec
    {:name ::inline-response-200-22
     :spec inline-response-200-22-data}))
